core/src/test/scala/unit/kafka/server/ConsumerGroupDescribeRequestsTest.scala

```diff
@@ -0,0 +1,175 @@
+    // in this test because it does not use FindCoordinator API.
+    createOffsetsTopic()
+
+    val admin = cluster.createAdminClient()
+    val topicId = TestUtils.createTopicWithAdminRaw(
+      admin = admin,
+      topic = "foo",
+      numPartitions = 3
+    )
+
+    val timeoutMs = 5 * 60 * 1000
+    val clientId = "client-id"
+    val clientHost = "/127.0.0.1"
+
+    // Add first group
+    val consumerGroupHeartbeatResponseData1 = consumerGroupHeartbeat(
+      groupId = "grp-1",
+      rebalanceTimeoutMs = timeoutMs,
+      subscribedTopicNames = List(""),
+      topicPartitions = List.empty
+    )
+
+    // Add second group
+    consumerGroupHeartbeat(
+      memberId = "member-1",
+      groupId = "grp-2",
+      rebalanceTimeoutMs = timeoutMs,
+      subscribedTopicNames = List("foo"),
+      topicPartitions = List.empty
+    )
+
+    // Add second member to second group
+    consumerGroupHeartbeat(
+      memberId = "member-2",
+      groupId = "grp-2",
+      rebalanceTimeoutMs = timeoutMs,
+      subscribedTopicNames = List("foo"),
+      topicPartitions = List.empty
+    )
+
+    val assignmentMember1 = new Assignment()
+      .setTopicPartitions(List(
+        new TopicPartitions()
+          .setTopicId(topicId)
+          .setTopicName("foo")
+          .setPartitions(List[Integer](0, 1, 2).asJava)
+      ).asJava)
+
+    val targetAssignmentMember1 = new Assignment()
+      .setTopicPartitions(List(
+        new TopicPartitions()
+          .setTopicId(topicId)
+          .setTopicName("foo")
+          .setPartitions(List[Integer](0, 1).asJava)
+      ).asJava)
+
+    val assignmentMember2 = new Assignment()
+      .setTopicPartitions(List().asJava)
+
+    val targetAssignmentMember2 = new Assignment()
+      .setTopicPartitions(List(
+        new TopicPartitions()
+          .setTopicId(topicId)
+          .setTopicName("foo")
+          .setPartitions(List[Integer](2).asJava)
+      ).asJava)
+
+    for (version <- ApiKeys.CONSUMER_GROUP_DESCRIBE.oldestVersion() to ApiKeys.CONSUMER_GROUP_DESCRIBE.latestVersion(isUnstableApiEnabled)) {
+
+      val expected = List(
+        new DescribedGroup()
+          .setGroupId("grp-1")
+          .setGroupState(ConsumerGroupState.STABLE.toString)
+          .setGroupEpoch(1)
+          .setAssignmentEpoch(1)
+          .setAssignorName("uniform")
+          .setMembers(List(
+            new ConsumerGroupDescribeResponseData.Member()
+              .setMemberId(consumerGroupHeartbeatResponseData1.memberId)
+              .setMemberEpoch(1)
+              .setClientId(clientId)
+              .setClientHost(clientHost)
+              .setSubscribedTopicRegex("")
+              .setSubscribedTopicNames(List("").asJava)
+          ).asJava),
+
+        new DescribedGroup()
+          .setGroupId("grp-2")
+          .setGroupState(ConsumerGroupState.RECONCILING.toString)
+          .setGroupEpoch(2)
+          .setAssignmentEpoch(2)
+          .setAssignorName("uniform")
```

Comment:
I have a minor ask here. Could we use the `range` assignor for this group to see whether the selection works as expected?

Reply:
Do you mean just using the range assignor here to see if the ConsumerGroupDescribe works, or actually testing the assignment of the range assignor. If the latter, this should be in another Test Class and probably also in another PR

Reply:
@riedelmax Do you mind doing this?

Reply:
@riedelmax sure

Reply:
@riedelmax I think that we could just do it here as we only need to change the assignor. What do you think?